webui/react/src/services/apiBuilder.ts

```diff
@@ -54,3 +59,22 @@ export function generateApi<Input, Output>(api: Api<Input, Output>) {
     }
   };
 }
+
+export const consumeStream = async <T = unknown>(
+  fetchArgs: DetSwagger.FetchArgs,
+  onEvent: (event: T) => void,
+  onFinish: () => void): Promise<void> => {
```

Comment:
suggestion: how about an optional `onError?: (error: Error) => void` that follows `onFinish` to be called in the `catch` block if available. This allows the consumer to handle errors.

Reply:
the `processApiError` will rethrow the exception so the consumer will always know and will be able to catch the error :thinking: would this onError eat the exception?

Reply:
I removed onFinish too because IMO it wasn't necessary. it's a async function we already have `.then and .catch` or the newer await syntax

Reply:
I like this better, seems much simpler!

Reply:
> the `processApiError` will rethrow the exception so the consumer will always know and will be able to catch the error 🤔 would this onError swallow the exception?

Since we don't have an `onError` probably makes sense to keep the rethrow in `processApiError` so the consumer can catch the error if it opts for it.